test/utils/test_convert.py

```diff
@@ -611,3 +612,231 @@ def test_from_dgl_hetero_graph():
+                                edge_type_attribute=None)
+
+    assert data["A"].x.tolist() == [4, 2]
+    assert data[("A", "to", "A")].edge_index.tolist() == [[0, 1], [1, 0]]
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_non_numeric_labels():
+    import networkx as nx
+
+    G = nx.Graph()
+    G.add_node('4', type='A')
+    G.add_node('2', type='A')
+    G.add_edge('4', '2')
+    for node in G.nodes():
+        G.nodes[node]['x'] = node
+
+    data = from_hetero_networkx(G, node_type_attribute="type",
+                                edge_type_attribute=None)
+
+    assert data["A"].x == ['4', '2']
+    assert data[("A", "to", "A")].edge_index.tolist() == [[0, 1], [1, 0]]
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_with_same_node_and_edge_attributes():
+    import networkx as nx
+
+    G = nx.Graph()
+    G.add_nodes_from([(0, {
+        'age': 1,
+        'type': 'A'
+    }), (1, {
+        'age': 6,
+        'type': 'A'
+    }), (2, {
+        'age': 5,
+        'type': 'A'
+    })])
+    G.add_edges_from([(0, 1, {'age': 2}), (1, 2, {'age': 7})])
+
+    data = from_hetero_networkx(G, node_type_attribute="type",
+                                edge_type_attribute=None)
+
+    assert data['A'].age.tolist() == [1, 6, 5]
+    assert data[('A', 'to', 'A')].edge_index.tolist() == [[0, 1, 1, 2],
+                                                          [1, 0, 2, 1]]
+    assert data[('A', 'to', 'A')].age.tolist() == [2, 2, 7, 7]
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_raise_missing_node_type_attribute():
+    import networkx as nx
+
+    G = nx.Graph()
+    G.add_node(0, type='A')
+    G.add_node(1)
+    G.add_edge(0, 1)
+
+    with pytest.raises(KeyError) as _:
+        from_hetero_networkx(G, node_type_attribute="type",
+                             edge_type_attribute=None)
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_raise_missing_edge_type_attribute():
+    import networkx as nx
+
+    G = nx.Graph()
+    G.add_node(0, type='A')
+    G.add_node(1, type='A')
+    G.add_edge(0, 1)
+
+    with pytest.raises(KeyError) as _:
+        from_hetero_networkx(G, node_type_attribute="type",
+                             edge_type_attribute="type")
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_raise_different_edge_attribute():
+    import networkx as nx
+
+    G = nx.Graph()
+    G.add_node(0, type='A')
+    G.add_node(1, type='A')
+    G.add_node(2, type='A')
+    G.add_edge(0, 1, a=1)
+    G.add_edge(0, 2, b=2)
+
+    with pytest.raises(ValueError) as _:
+        from_hetero_networkx(G, node_type_attribute="type",
+                             edge_type_attribute=None)
+
+
+@withPackage('networkx')
+def test_from_hetero_networkx_raise_different_node_attribute():
+    import networkx as nx
+
+    G = nx.Graph()
+    G.add_node(0, type='A', a=1)
+    G.add_node(1, type='A', b=1)
+
+    with pytest.raises(ValueError) as _:
```

Comment:
```suggestion
    with pytest.raises(ValueError, match = r''Not all nodes .* attributes.'):
```
Since `ValueError` is common and could be raised for other reasons, its important to check if the expected value error is raised. So please use `match` here and in other tests.